src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -306,6 +322,24 @@ private void fireNextRequest() {
         PatternsCS.pipe(responsePromise, context().dispatcher()).to(self());
     }
 
+    private CompletableFuture<Response> sendHttpRequest(
+            final Request request,
+            final int attempt) {
+        final CompletableFuture<Response> promise = new CompletableFuture<>();
+        _client.executeRequest(request, new ResponseAsyncCompletionHandler(promise));
+        promise.handle((result, err) -> {
+            if (err == null && ACCEPTED_STATUS_CODES.contains(result.getStatusCode())) {
+                try (Metrics metrics = _metricsFactory.create()) {
```

Comment:
Creating a new Metrics object for each POST is going to be quite expensive

Reply:
Yes, it's not ideal; unfortunately, there is no end-to-end scoped instrumentation in CAGG. Not to mention the event loop here is async and so decoupled. If CAGG supports periodic metrics, we could use that, but this is not unreasonable (there's an MF in the Router so for each _incoming_ request; [Ref](https://github.com/ArpNetworking/metrics-cluster-aggregator/blob/master/src/main/java/com/arpnetworking/clusteraggregator/http/Routes.java#L102)).